NUCLEO_G031K8 Failed to build all greentea images in bare-metal profile

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->
after  NUCLEO_G031K8 been added to mbed-os via https://github.com/ARMmbed/mbed-os/pull/13006 , we found that  NUCLEO_G031K8 failed on building greentea `TESTS-MBED_HAL-MINIMUM_REQUIREMENTS`
#### Target(s) affected by this defect ?
 NUCLEO_G031K8

#### Toolchain(s) (name and version) displaying this defect ?
GCC_ARM only, ARMC6 works fine

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->
latest master : afcf91f3314d82ce8d555f4a891c9fd8db13910f

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)
mbed-cli

#### How is this defect reproduced ? 
run:
```
mbed test --compile -m NUCLEO_G031K8 -t GCC_ARM --app-config TESTS/configs/baremetal.json
```
Error:
```
Build failures:
  * NUCLEO_G031K8::GCC_ARM::TESTS-MBED_HAL-MINIMUM_REQUIREMENTS
        Building project minimum_requirements (NUCLEO_G031K8, GCC_ARM)
        Scan: GCC_ARM
        Scan: minimum_requirements
        Compile [100.0%]: main.cpp
        Link: minimum_requirements
        /usr/local/gcc-arm-none-eabi-9-2019-q4-major/bin/../lib/gcc/arm-none-eabi/9.2.1/../../../../arm-none-eabi/bin/ld:./BUILD/tests/NUCLEO_G031K8/GCC_ARM/./TESTS/mbed_hal/minimum_requirements/.link_script.ld:90 cannot move location counter backwards (from 00000000200010b8 to 0000000020001000)
        collect2: error: ld returned 1 exit status
        
[mbed] Working path "/builds/ws/mbed-os-ci_build-GCC_ARM@3/mbed-os" (program)
[mbed] WARNING: Could not find mbed program in current path "/builds/ws/mbed-os-ci_build-GCC_ARM@3/mbed-os".
       You can fix this by calling "mbed new ." in the root of your program.
---
[mbed] ERROR: "/usr/bin/python" returned error.
       Code: 1
       Path: "/builds/ws/mbed-os-ci_build-GCC_ARM@3/mbed-os"
       Command: "/usr/bin/python -u /builds/ws/mbed-os-ci_build-GCC_ARM@3/mbed-os/tools/test.py -D MBED_TRAP_ERRORS_ENABLED=1 -D SKIP_TIME_DRIFT_TESTS=1 -D MBED_TEST_MODE -t GCC_ARM -m NUCLEO_G031K8 --source . --build ./BUILD/tests/NUCLEO_G031K8/GCC_ARM --test-spec ./BUILD/tests/NUCLEO_G031K8/GCC_ARM/test_spec.json --build-data BUILD/tests/build_data.json --app-config TESTS/configs/baremetal.json --greentea"
       Tip: You could retry the last command with "-v" flag for verbose output
```

--- a/targets/targets.json
+++ b/targets/targets.json
@@ -2374,6 +2374,12 @@
             "MCU_STM32G0"
         ],
         "public": false,
+        "overrides": {
+            "boot-stack-size": "0x400"
+        },
+        "supported_application_profiles": [
+            "bare-metal"
+        ],
         "extra_labels_add": [
             "STM32G031xx"
         ],
@@ -2388,24 +2394,7 @@
         "detect_code": [
             "0852"
         ],
-        "device_name": "STM32G031K8Tx",
-        "supported_c_libs": {
-            "arm": [
-                "std",
-                "small"
-            ],
-            "gcc_arm": [
-                "std",
-                "small"
-            ],
-            "iar": [
-                "std"
-            ]
-        },
-        "c_lib": "small",
-        "supported_application_profiles": [
-            "bare-metal"
-        ]
+        "device_name": "STM32G031K8Tx"
     },
     "MCU_STM32G071xx": {
         "inherits": [

